pkgs/dart_tooling_mcp_server/lib/src/mixins/dtd.dart

```diff
@@ -678,28 +791,19 @@ extension type DebugSession.fromJson(Map<String, Object?> _value)
   String get id => _value['id'] as String;
   String get name => _value['name'] as String;
   String get projectRootPath => _value['projectRootPath'] as String;
-  String get vmServiceUri => _value['vmServiceUri'] as String;
+  String? get vmServiceUri => _value['vmServiceUri'] as String?;
 
   factory DebugSession({
     required String debuggerType,
     required String id,
     required String name,
     required String projectRootPath,
-    required String vmServiceUri,
+    required String? vmServiceUri,
   }) => DebugSession.fromJson({
     'debuggerType': debuggerType,
     'id': id,
     'name': name,
     'projectRootPath': projectRootPath,
-    'vmServiceUri': vmServiceUri,
+    if (vmServiceUri != null) 'vmServiceUri': vmServiceUri,
```

Comment:
what type of debug sessions do not have a vm service uri?

Reply:
When we get the `started` notification, they don' that's a VM service URI yet. Then we get a follow-up `changed` notification that does contain the VM service URI.